### PR TITLE
WIP - New Developer Guide sans docstrings

### PR DESCRIPTION
**Pull request by mhoeber**

### CODE REVIEW

**Comment by mhoeber** [comment]
@nedbat   This is branch to move/separat developer doc.  I can better articulate benefits around quality/process improvement and control, sharing content across other guides (JS Input is a good example here), versioning for open edx, building better tests and publishing automation, and more.

Draft is at:  http://draft-dev-guide-sans-docstrings.readthedocs.org/en/latest/index.html

That's largely the same content, slight reorg to start compared to:  http://edx.readthedocs.org/projects/edx-developer-guide/en/latest/

@srpearce  can you sanity check new version

**Comment by srpearce** [comment]
:+1:

**Comment by mhoeber** [comment]
@singingwolfboy @sarina

**Comment by singingwolfboy** [comment]
Please don't commit `*.pyc` files to the repository. You should exclude them in `.gitignore`, so that you don't accidentally add them to commits.

File: en_us/developers/source/process/code-considerations.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ Operational Impact
   provision anything new when new courses launch? when new schools start? etc....
 * Has the feature been tested using a production configuration with vagrant?
 
-See also: :doc:`deploy-new-service`
+See also: :ref:`Deploy a New Service`.
```
Comment (singingwolfboy) [inline en_us/developers/source/process/code-considerations.rst:40]: Why did you change this from a `:doc:` reference to a `:ref:` reference?

Comment (mhoeber) [inline en_us/developers/source/process/code-considerations.rst:40]: easier maintenance if things move around

**Comment by mhoeber** [comment]
@doctoryes @dmitchell @cpennington

**Comment by sarina** [comment]
@mhoeber I'm not sure how you want me to review this - I don't fully understand how all the references, etc work. And it would be good to get the .pyc files out of your diff to reduce clutter.

**Comment by mhoeber** [comment]
@sarina   no specific expectations, just trying to keep folks aware of this change.  I will definitely get the pyc files out.
